src/drivers/drv_adc.h

```diff
@@ -42,14 +42,15 @@
 
 #include <stdint.h>
 #include <sys/ioctl.h>
+#include <uORB/topics/adc_report.h>
 
 /* Define the PX4 low level format ADC and the maximum
  * number of channels that can be returned by a lowlevel
  * ADC driver. Drivers may return less than PX4_MAX_ADC_CHANNELS
  * but no more than PX4_MAX_ADC_CHANNELS.
  *
  */
-#define PX4_MAX_ADC_CHANNELS 12
+#define PX4_MAX_ADC_CHANNELS ((int)(sizeof(adc_report_s::channel_id)/sizeof(*adc_report_s::channel_id)))
```

Comment:
This looks wrong:
(sizeof(adc_report_s::channel_id)/sizeof(*adc_report_s::channel_id[0])))

There is an ArraySize Macro in the tree px4_macros.h. IMHO I think code using ArraySize(adc_report_s::channel_id)) is better then having the #define (That define is needed in the ADC driver as a constant.  - then to insure the the ADC driver has enough channels a static assert should be used as @bkueng suggested.

Reply:
Thanks for advice. I'm using `arraySize()` instead of that. But is it necessary to apply a static assert? I don't agree with that, because the max number of available channel is only defined in `adc_report.msg` now.